### PR TITLE
Fix bug where coredns 1.8.3 needs more permissions

### PR DESCRIPTION
**Pull request by yorinasub17**
Fixes https://github.com/gruntwork-io/kubergrunt/issues/129

EKS integration testing PR is https://github.com/gruntwork-io/terraform-aws-eks/pull/331

### CODE REVIEW

File: eks/sync.go
Comment (rhoboat) [inline eks/sync.go:259]: NIT: I would probably word this comment as 
> The keyword "upstream" or \`upstream\`

to be clear that `upstream` is the keyword.

File: eks/sync.go
Comment (rhoboat) [inline eks/sync.go:265]: Since annotated versions are considered `<`, the comparison is actually against `1.7.0-eksbuild.1`, which is `< 1.7.0`. I don't know if it really matters in this comment, but it strikes me as slightly inaccurate. Do I understand correctly?

Comment (rhoboat) [reply eks/sync.go:265]: I think it could be argued that it doesn't matter for the sake of this comment, since `1.7.0-eksbuild.1` is still `1.7.0` colloquially.

**Review by rhoboat — APPROVED** [review]
LGTM! Left minor nits, feel free to take or leave.

NIT: On the whole, I wonder about these updates scaling. I like that these two new `update...Compatibility` functions are named for both the version and the thing being updated. I wonder if it is easier in the long run to process compatibility updates for coredns in a slightly different way, where maybe, based on a single semverStringCompare, you run all the updates that are needed. Again, definitely not critical now.

File: eks/sync.go
Comment (yorinasub17) [reply eks/sync.go:265]: This doesn't really matter, because all the EKS coredns versions are always published with the annotation, so the comment is accurate in the context of EKS.

File: eks/sync.go
Comment (yorinasub17) [reply eks/sync.go]: Done! 7cb60c0

**Comment by yorinasub17** [comment]
> NIT: On the whole, I wonder about these updates scaling. I like that these two new update...Compatibility functions are named for both the version and the thing being updated. I wonder if it is easier in the long run to process compatibility updates for coredns in a slightly different way, where maybe, based on a single semverStringCompare, you run all the updates that are needed. Again, definitely not critical now.

I don't think we should worry too much about this. EKS is only going to maintain support for a handful of kubernetes versions (as evidenced by the fact that they continue to roll off the oldest version everytime they add support for a new version). And the number of versions that need special treatment will be even less than that.

Furthermore, this command is going to be deprecated real soon when we switch to using managed add-ons, which will encapsulate this logic at the AWS API level.

> I wonder if it is easier in the long run to process compatibility updates for coredns in a slightly different way, where maybe, based on a single semverStringCompare, you run all the updates that are needed.

**Review by rhoboat — APPROVED** [review]
LGTM!

**Comment by yorinasub17** [comment]
> I wonder if it is easier in the long run to process compatibility updates for coredns in a slightly different way, where maybe, based on a single semverStringCompare, you run all the updates that are needed. 

Can you elaborate what you are thinking here? I don't think I fully understand what you are proposing here. I'm having a hard time figuring out how you can reduce the checks down to a single `semverStringCompare` given the number of path ways that one updates their kubernetes cluster (and thus you can end up with needing to run both updates because you are hopping from <1.7.0 to >1.8.3).

**Comment by yorinasub17** [comment]
Thanks for the review! I'm waiting on the integration testing build to pass before merging this in.

**Comment by rhoboat** [comment]
> > I wonder if it is easier in the long run to process compatibility updates for coredns in a slightly different way, where maybe, based on a single semverStringCompare, you run all the updates that are needed.
> 
> Can you elaborate what you are thinking here? I don't think I fully understand what you are proposing here. I'm having a hard time figuring out how you can reduce the checks down to a single `semverStringCompare` given the number of path ways that one updates their kubernetes cluster (and thus you can end up with needing to run both updates because you are hopping from <1.7.0 to >1.8.3).

Good question. I was thinking the way to know which (how many) updates to apply would be derived from the difference in versions. 🤔 I am not sure if that's possible. I think the `.Compare` function might only return `-1, 0, 1`.

**Comment by yorinasub17** [comment]
> Good question. I was thinking the way to know which (how many) updates to apply would be derived from the difference in versions. 🤔 I am not sure if that's possible. I think the .Compare function might only return -1, 0, 1.

**Comment by yorinasub17** [comment]
> Good question. I was thinking the way to know which (how many) updates to apply would be derived from the difference in versions. 🤔 I am not sure if that's possible. I think the .Compare function might only return -1, 0, 1.

Wouldn't that be much more confusing as a maintainer? E.g., you now have to do the math to update the `-N` number everytime a new backward incompatible update is released.

**Comment by yorinasub17** [comment]
Ok integration testing branch passed, so will go ahead and merge this in! Thanks for the review!

**Pull request merged**
